controllers/vspheredeploymentzone_controller.go

```diff
@@ -147,12 +147,17 @@ func (r vsphereDeploymentZoneReconciler) Reconcile(ctx goctx.Context, request re
 		return ctrl.Result{}, r.reconcileDelete(vsphereDeploymentZoneContext)
 	}
 
+	// If the VSphereDeploymentZone doesn't have our finalizer, add it.
+	// Requeue immediately after adding finalizer to avoid the race condition between init and delete
+	if !ctrlutil.ContainsFinalizer(vsphereDeploymentZone, infrav1.DeploymentZoneFinalizer) {
+		ctrlutil.AddFinalizer(vsphereDeploymentZoneContext.VSphereDeploymentZone, infrav1.DeploymentZoneFinalizer)
```

Comment:
```suggestion
		ctrlutil.AddFinalizer(vsphereDeploymentZone, infrav1.DeploymentZoneFinalizer)
```
nit: just for consistency with the line above

Reply:
Will fix it next patch